scrapy/shell.py

```diff
@@ -159,7 +156,7 @@ def get_help(self):
         return "\n".join("[s] %s" % l for l in b)
 
     def _is_relevant(self, value):
-        return isinstance(value, self.relevant_classes)
+        return isinstance(value, self.relevant_classes) or is_item_like(value)
```

Comment:
Looking at line 122 (old 125), I think this file does not need any change, given that `_is_relevant` is only used with the contents of `self.vars`.

Reply:
Actually it seems to be necessary for the item to be displayed in the list of variables. It's just an aesthetic thing, since the variable is populated nonetheless, just not displayed.

```python
# inventory.py
from dataclasses import dataclass, field

@dataclass
class InventoryItem:
    name: list = field(default_factory=list)
    price: list = field(default_factory=list)
    stock: list = field(default_factory=list)
```

```
$ scrapy shell https://example.org -s DEFAULT_ITEM_CLASS=inventory.InventoryItem
2019-11-29 13:37:40 [scrapy.utils.log] INFO: Scrapy 1.8.0 started (bot: scrapybot)
(...)
[s] Available Scrapy objects:
[s]   scrapy     scrapy module (contains scrapy.Request, scrapy.Selector, etc)
[s]   crawler    <scrapy.crawler.Crawler object at 0x7fcd90b1cbe0>
[s]   item       InventoryItem(name=[], price=[], stock=[])
[s]   request    <GET https://example.org>
[s]   response   <200 https://example.org>
[s]   settings   <scrapy.settings.Settings object at 0x7fcd8c94d7f0>
[s]   spider     <DefaultSpider 'default' at 0x7fcd8b668f60>
[s] Useful shortcuts:
[s]   fetch(url[, redirect=True]) Fetch URL and update local objects (by default, redirects are followed)
[s]   fetch(req)                  Fetch a scrapy.Request and update local objects 
[s]   shelp()           Shell help (print this help)
[s]   view(response)    View response in a browser
In [1]: item
Out[1]: InventoryItem(name=[], price=[], stock=[])
```

For reference, here is where the `item` variable gets populated: https://github.com/scrapy/scrapy/blob/1.8.0/scrapy/shell.py#L125